plugins/engine-configuration/impl/src/main/java/org/pentaho/di/engine/configuration/impl/RunConfigurationProviderFactoryManagerImpl.java

```diff
@@ -0,0 +1,79 @@
+/*! ******************************************************************************
+ *
+ * Pentaho
+ *
+ * Copyright (C) 2024 by Hitachi Vantara, LLC : http://www.pentaho.com
+ *
+ * Use of this software is governed by the Business Source License included
+ * in the LICENSE.TXT file.
+ *
+ * Change Date: 2029-07-20
+ ******************************************************************************/
+
+package org.pentaho.di.engine.configuration.impl;
+
+import org.pentaho.di.core.exception.KettlePluginException;
+import org.pentaho.di.core.service.PluginServiceLoader;
+import org.pentaho.di.engine.configuration.api.CheckedMetaStoreSupplier;
+import org.pentaho.di.engine.configuration.api.RunConfigurationProvider;
+import org.pentaho.di.engine.configuration.api.RunConfigurationProviderFactory;
+import org.pentaho.di.engine.configuration.api.RunConfigurationProviderFactoryManager;
+import org.pentaho.di.engine.configuration.impl.pentaho.DefaultRunConfigurationProviderFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * The RunConfigurationProviderFactoryManager is used to manage the registration of RunConfiguration types and
+ * generate their RunConfigurationProviders for RunConfigurationManager instances associated with Projects
+ * <p>
+ * Each RunConfiguration type (e.g. Pentaho, Spark) has a RunConfigurationProvider that allows users to create
```

Comment:
I wouldn't explicitly mention "Spark", this is CE code and spark is not a concept or plugin in this domain. Who knows what the future could hold, "Spark" might not even be an option.